modules/reactor/src/meshgenerators/CoreMeshGenerator.C

```diff
@@ -0,0 +1,427 @@
+                             "of the assembly-like structures is defined, and "
+                             "enforced by declaration in the ReactorMeshParams.");
+
+  return params;
+}
+
+CoreMeshGenerator::CoreMeshGenerator(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _inputs(getParam<std::vector<MeshGeneratorName>>("inputs")),
+    _empty_key(getParam<MeshGeneratorName>("dummy_assembly_name")),
+    _pattern(getParam<std::vector<std::vector<unsigned int>>>("pattern")),
+    _extrude(getParam<bool>("extrude"))
+{
+  /* Below, we get the ReactorMeshParams info from one of the pin inputs.
+     The reactor params gives the axial info if we are to extrude as well as the assembly pitch
+     that the pin pattern is to be checked against.
+     We then create the PatternedMesh subgenerators and pass the parameter info needed for the
+     inputs. Finally, if the the mesh is to be extruded we create the FancyExtruderGenerator
+     subgenerator and the PlaneIDMeshGenerator subgenerator to handle this.
+  */
+  MeshGeneratorName _reactor_params =
+      MeshGeneratorName(getMeshProperty<std::string>("reactor_params_name", _inputs[0]));
+
+  // Ensure that the user has supplied the correct info for conformal mesh generation
+  if (!hasMeshProperty("mesh_dimensions", _reactor_params) ||
+      !hasMeshProperty("mesh_geometry", _reactor_params))
+    mooseError("The reactor_params input must be a ReactorMeshParams type MeshGenerator\n");
+  else
+  {
+    _geom_type = getMeshProperty<std::string>("mesh_geometry", _reactor_params);
+    _mesh_dimensions = getMeshProperty<int>("mesh_dimensions", _reactor_params);
+  }
+  if (_extrude && getMeshProperty<unsigned int>("mesh_dimensions", _reactor_params) != 3)
+    mooseError("This is a 2 dimensional mesh, you cannot extrude it. Check your ReactorMeshParams "
+               "inputs\n");
+  if (_extrude && (!hasMeshProperty("top_boundary_id", _reactor_params) ||
+                   !hasMeshProperty("bottom_boundary_id", _reactor_params)))
+    mooseError("Both top_boundary_id and bottom_boundary_id must be provided in ReactorMeshParams "
+               "if using extruded geometry");
+
+  std::size_t empty_pattern_loc = 0;
+  bool make_empty = false;
+  for (auto assembly : _inputs)
+  {
+    if (assembly != _empty_key)
+    {
+      ++empty_pattern_loc;
+      if (getMeshProperty<bool>("extruded", assembly))
+        mooseError("Assemblies that have already been extruded cannot be used in CoreMeshGenerator "
+                   "definition.\n");
+    }
+    else
+    {
+      make_empty = true;
+      for (const auto i : make_range(_pattern.size()))
+      {
+        for (const auto j : make_range(_pattern[i].size()))
+        {
+          if (_pattern[i][j] == empty_pattern_loc)
+            _empty_pos = true;
+        }
+      }
+    }
+  }
+  //***Add checks for pins to be the same size and pin sizes to add to Assembly pitch
```

Comment:
what s the status on this one?

Reply:
These checks now occur in AssemblyMeshGenerator.